c10/util/TypeCast.h

```diff
@@ -21,7 +21,7 @@ C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-int-float-conversion")
 namespace c10 {
 
 template <typename dest_t, typename src_t>
-struct needs_real {
+struct needs_real_or_bool {
```

Comment:
Here "real" means "not complex", as `dest_t` may also be an int, so no need to change the name really.

Reply:
Here I took the name to refer to "needs real from" `src_t` but let me change it back.